aten/src/THC/THCTensorMathReduce.cuh

```diff
@@ -105,21 +105,26 @@ struct SquareFunctor<ResT, half> {
 template <typename T>
 struct ReduceMin {
   inline __device__ T operator()(T a, T b) const {
-    return THCNumerics<T>::lt(a, b) ? a : b;
+    // a != a means a == NaN
+    return (THCNumerics<T>::lt(a, b) ||
```

Comment:
I'd prefer an inline function that does this instead of having this comment everywhere below.

Reply:
Something like isnan can go to THCNumerics. There's isnan in cuda for floats and doubles, but not for halfs.